doc/en/assert.txt

```diff
@@ -113,6 +113,38 @@ exceptions your own code is deliberately raising, whereas using
 like documenting unfixed bugs (where the test describes what "should" happen)
 or bugs in dependencies.
 
+Assertions about expected warnings
+-----------------------------------------
+
+.. versionadded:: 2.8
+
+You can also check that code raises a particular warning using ``pytest.warns``,
+which works in a similar manner to ``pytest.raises`` (see above)::
+
+    import warnings
+    import pytest
+
+    def test_warning():
+        with pytest.warns(UserWarning):
+            warnings.warn("my warning", UserWarning)
+
+The test will fail if the warning in question is not raised. The function also
+returns a list of all raised warnings (as ``warnings.WarningMessage`` objects),
+which you can query for additional information::
+
+    with pytest.warns(RuntimeWarning) as record:
+        warnings.warn("another warning", RuntimeWarning)
+
+    assert len(record) == 1  # check that only one warning was raised
+    assert record[0].message.message == "another warning"
```

Comment:
No, that's actually the way you have to do it, because of how `warnings.WarningMessage` is set up. `WarningMessage.message` returns the actual exception, not the message, so you have to call `message` again. Though your question made me realize that in Python3, warnings no longer have a `message` attribute, just an `args` one, so I've changed things to work in Python3 as well.